client/me/account/main.jsx

```diff
@@ -663,6 +684,85 @@ const Account = createReactClass( {
 		);
 	},
 
+	handleSubmitError( error, formName = '' ) {
+		debug( 'Error saving settings: ' + JSON.stringify( error ) );
+
+		// handle error case here
+		if ( error.message ) {
+			this.props.errorNotice( error.message, noticeOptions );
+		} else {
+			this.props.errorNotice(
+				this.props.translate( 'There was a problem saving your changes.' ),
+				noticeOptions
+			);
+		}
+
+		this.setState( {
+			submittingForm: false,
+			formsSubmitting: {
+				...this.state.formsSubmitting,
+				...( formName && { [ formName ]: false } ),
+			},
+		} );
+	},
+
+	isSubmittingForm( formName ) {
+		return formName ? this.state.formsSubmitting[ formName ] : this.state.submittingForm;
+	},
+
+	handleSubmitSuccess( response, formName = '' ) {
+		this.props.markSaved && this.props.markSaved();
+
+		if ( this.state && this.state.redirect ) {
+			user()
+				.clear()
+				.then( () => {
+					// Sometimes changes in settings require a url refresh to update the UI.
+					// For example when the user changes the language.
+					window.location = this.state.redirect + '?updated=success';
+				} );
+			return;
+		}
+		// if we set submittingForm too soon the UI updates before the response is handled
+		this.setState(
+			{
+				submittingForm: false,
+				formsSubmitting: {
+					...this.state.formsSubmitting,
+					...( formName && { [ formName ]: false } ),
+				},
+			},
+			() => {
+				this.props.successNotice(
+					this.props.translate( 'Settings saved successfully!' ),
+					noticeOptions
+				);
+			}
+		);
+		debug( 'Settings saved successfully ' + JSON.stringify( response ) );
+	},
+
+	submitForm: function ( event, settingsToSave, formName = '' ) {
```

Comment:
Is there a chance `formName` will be empty? It feels odd to ever have such a case. Can we always consider it defined? That way we'll be able to avoid checks like `formName &&` below.